src/test/integration/integration-test.py

```diff
@@ -25,50 +25,64 @@ def test_get_authority_onids(self, endpoint='/spendingauthority'):
+    def checking_spending_limits(self, attributes):
         # Checking that the lists of spending limits and indexes are unique
         # Checking valid onids with spending authority only
-        # To make sure that attributes is not empty
-        response_data = response.json()['data']
-        attributes = response_data['attributes']
-        self.assertTrue(attributes)
-        limits = attributes['limits']
-        # Checking that limits has at least one index
-        logging.debug(f'limits: {len(limits)}')
-        self.assertTrue(limits)
-        spendinglimit = []
+        # To make sure that attributes and index are not empty
+        non_empty_list_dict = {
+            'attributes': attributes,
+            'limits': attributes['limits']
+        }
+
+        for key, non_empty_list in non_empty_list_dict.items():
+            logging.debug(f'{key}: {len(non_empty_list)}')
+            self.assertTrue(non_empty_list)
+
+        spending_limit = []
         index_array = []
-        for limit in limits:
-            for index in limit['indexes']:
-                index_array.append(index)
-            spendinglimit.append(limit['spendingLimit'])
-            # Checking that each spending limit has at least one index
-            logging.debug(f'spending limit has: {limit["indexes"]}')
-            self.assertTrue(limit['indexes'])
+
+        for limit in non_empty_list_dict['limits']:
+            try:
+                for index in limit['indexes2']:
+                    index_array.append(index)
+                # Checking that each spending limit has at least one index
+                logging.debug(f'spending limit has: {limit["indexes"]}')
+                self.assertTrue(limit['indexes'])
+                try:
+                    spending_limit.append(limit['spendingLimit'])
+                except KeyError as error:
+                    logging.debug('spendingLimit does not exist')
+                    self.fail(error)
+            except KeyError as error:
+                logging.debug('indexes does not exist')
+                self.fail(error)
+
         # Comparing the size of the list with the size of the set of that
         # list doublications will be removed in a the set
-        logging.debug(f'index_array size: {len(index_array)}')
-        logging.debug(f'index_array set size: {len(set(index_array))}')
-        self.assertEqual(len(index_array), len(set(index_array)))
-        logging.debug(f'index_array size:{len(spendinglimit)}')
-        logging.debug(f'index_array set size:{len(set(spendinglimit))}')
-        self.assertEqual(len(spendinglimit), len(set(spendinglimit)))
+        equal_list_dict = {
+            'index_array': index_array,
+            'spending_limit': spending_limit
+        }
+
+        for key, equal_list in equal_list_dict.items():
+            logging.debug(f'{key} size: {len(equal_list)}')
+            logging.debug(f'{key} set size: {len(set(equal_list))}')
```

Comment:
`assertEqual` below will give you almost exactly the same info as your debug message here if the assertion failed, so I don't think the debug message here is necessary.